commit 9437b3b233d61b8a7da0b2056b026e72bf9d5194
Author: Naseem <naseemkullah@gmail.com>
Date:   Wed Mar 13 20:37:10 2019 -0400

    [stable/datadog] Add fullnameOverride and nameOverride in README and values file (#11903)
    
    * Refactor naming
    
    Add fullnameOverride and nameOverride in README+values file
    Use recommended labels
    Refactor naming of objects
    
    Signed-off-by: Naseem Ullah <naseemkullah@gmail.com>
    
    * Modify README and values
    
    Signed-off-by: Naseem Ullah <naseemkullah@gmail.com>

diff --git a/stable/datadog/Chart.yaml b/stable/datadog/Chart.yaml
index 1f0fc8353..39f65dba3 100755
--- a/stable/datadog/Chart.yaml
+++ b/stable/datadog/Chart.yaml
@@ -1,5 +1,5 @@
 name: datadog
-version: 1.22.0
+version: 1.23.0
 appVersion: 6.9.0
 description: DataDog Agent
 keywords:
diff --git a/stable/datadog/README.md b/stable/datadog/README.md
index 3b71f1385..f89f714ab 100644
--- a/stable/datadog/README.md
+++ b/stable/datadog/README.md
@@ -35,7 +35,7 @@ However, you can use manually created secret by setting the `datadog.apiKeyExist
 
 Read about the Datadog Cluster Agent in the [official documentation](https://docs.datadoghq.com/agent/kubernetes/cluster/).
 
-Run the following if you want to deploy the chart with the Datadog Cluster Agent: 
+Run the following if you want to deploy the chart with the Datadog Cluster Agent:
 
 ```bash
 helm install --name datadog-monitoring \
@@ -53,22 +53,24 @@ The Leader Election is enabled by default in the chart for the Cluster Agent. On
 
 #### Cluster Agent Token
 
-You can specify the Datadog Cluster Agent token used to secure the communication between the Cluster Agent(s)q and the Agents with `clusterAgent.token`. 
+You can specify the Datadog Cluster Agent token used to secure the communication between the Cluster Agent(s)q and the Agents with `clusterAgent.token`.
 
 **If you don't specify a token, a random one is generated at each deployment so you must use `--recreate-pods` to ensure all pod use the same token.** see[Datadog Chart notes](https://github.com/helm/charts/blob/57d3030941ad2ec2d6f97c86afdf36666658a884/stable/datadog/templates/NOTES.txt#L49-L59) to learn more.
 
-### Updating
+### Upgrading
 
-Starting with version 1.0.0, this chart does not support deploying Agent 5.x anymore. If you cannot upgrade to Agent 6.x, you can use a previous version of the chart by calling helm install with `--version 0.18.0`.
-
-See [0.18.1's README](https://github.com/helm/charts/blob/847f737479bb78d89f8fb650db25627558fbe1f0/stable/datadog/README.md) to see which options were supported at the time.
-
-#### From < 1.19.0 to >= 1.19.0
+#### From 1.19.0 onwards
 
 Version `1.19.0` introduces the use of release name as full name if it contains the chart name(`datadog` in this case).
 E.g. with a release name of `datadog`, this renames the `DaemonSet` from `datadog-datadog` to `datadog`.
 The suggested approach is to delete the release and reinstall it.
 
+#### From 1.0.0 onwards
+
+Starting with version 1.0.0, this chart does not support deploying Agent 5.x anymore. If you cannot upgrade to Agent 6.x, you can use a previous version of the chart by calling helm install with `--version 0.18.0`.
+
+See [0.18.1's README](https://github.com/helm/charts/blob/847f737479bb78d89f8fb650db25627558fbe1f0/stable/datadog/README.md) to see which options were supported at the time.
+
 ### Uninstalling the Chart
 
 To uninstall/delete the `<RELEASE_NAME>` deployment:
@@ -81,7 +83,7 @@ The command removes all the Kubernetes components associated with the chart and
 
 ## Configuration
 
-As a best practice, a YAML file that specifies the values for the chart parameters should be provided to configure the chart: 
+As a best practice, a YAML file that specifies the values for the chart parameters should be provided to configure the chart:
 
 1.  **Copy the default [`datadog-values.yaml`](/values.yaml) value file.**
 2.  Set the `apiKey` parameter with your [Datadog API key](https://app.datadoghq.com/account/settings#api).
@@ -112,7 +114,7 @@ helm upgrade -f datadog-values.yaml <RELEASE_NAME> stable/datadog --recreate-pod
 
 ### Enabling Process Collection
 
-Update your [datadog-values.yaml](/values.yaml) file with the process collection configuration: 
+Update your [datadog-values.yaml](/values.yaml) file with the process collection configuration:
 
 ```
 datadog:
@@ -134,9 +136,9 @@ Alternatively set the `datadog.leaderElection`, `datadog.collectEvents` and `rba
 
 ### conf.d and checks.d
 
-The Datadog [entrypoint](https://github.com/DataDog/datadog-agent/blob/master/Dockerfiles/agent/entrypoint/89-copy-customfiles.sh) copies files with a `.yaml` extension found in `/conf.d` and files with `.py` extension in `/check.d` to `/etc/datadog-agent/conf.d` and `/etc/datadog-agent/checks.d` respectively. 
+The Datadog [entrypoint](https://github.com/DataDog/datadog-agent/blob/master/Dockerfiles/agent/entrypoint/89-copy-customfiles.sh) copies files with a `.yaml` extension found in `/conf.d` and files with `.py` extension in `/check.d` to `/etc/datadog-agent/conf.d` and `/etc/datadog-agent/checks.d` respectively.
 
-The keys for `datadog.confd` and `datadog.checksd` should mirror the content found in their respective ConfigMaps. Update your [datadog-values.yaml](/values.yaml) file with the check configurations: 
+The keys for `datadog.confd` and `datadog.checksd` should mirror the content found in their respective ConfigMaps. Update your [datadog-values.yaml](/values.yaml) file with the check configurations:
 
 ```yaml
 datadog:
@@ -173,7 +175,7 @@ For more details, please refer to [the documentation](https://docs.datadoghq.com
 
 ### Kubernetes Labels and Annotations
 
-To map Kubernetes pod labels and annotations to Datadog tags, provide a dictionary with kubernetes labels/annotations as keys and Datadog tags key as values in your [datadog-values.yaml](/values.yaml) file: 
+To map Kubernetes pod labels and annotations to Datadog tags, provide a dictionary with kubernetes labels/annotations as keys and Datadog tags key as values in your [datadog-values.yaml](/values.yaml) file:
 
 ```yaml
 podAnnotationsAsTags:
@@ -220,6 +222,8 @@ helm install --name <RELEASE_NAME> \
 | `image.tag`                              | The image tag to pull                                                                     | `6.9.0`                                     |
 | `image.pullPolicy`                       | Image pull policy                                                                         | `IfNotPresent`                              |
 | `image.pullSecrets`                      | Image pull secrets                                                                        | `nil`                                       |
+| `nameOverride`                           | Override name of app                                                                      | `nil`                                       |
+| `fullnameOverride`                       | Override full name of app                                                                 | `nil`                                       |
 | `rbac.create`                            | If true, create & use RBAC resources                                                      | `true`                                      |
 | `rbac.serviceAccount`                    | existing ServiceAccount to use (ignored if rbac.create=true)                              | `default`                                   |
 | `datadog.name`                           | Container name if Daemonset or Deployment                                                 | `datadog`                                   |
diff --git a/stable/datadog/templates/NOTES.txt b/stable/datadog/templates/NOTES.txt
index 9cb829aeb..ef6d1e6f3 100644
--- a/stable/datadog/templates/NOTES.txt
+++ b/stable/datadog/templates/NOTES.txt
@@ -89,7 +89,7 @@ chart if your use case still requires a separate Deployment.
 
 The autoconf value is deprecated, Autodiscovery templates can now
 be safely moved to the confd value. As a temporary measure, both
-values were merged into the {{ template "datadog.confd.fullname" . }} configmap,
+values were merged into the {{ template "datadog.fullname" . }}-confd configmap,
 but this will be removed in a future chart release.
 Please note that duplicate file names may have conflicted during
 the merge. In that case, the confd entry will take precedence.
diff --git a/stable/datadog/templates/_helpers.tpl b/stable/datadog/templates/_helpers.tpl
index 2dddf3d36..4315a4245 100644
--- a/stable/datadog/templates/_helpers.tpl
+++ b/stable/datadog/templates/_helpers.tpl
@@ -24,6 +24,13 @@ If release name contains chart name it will be used as a full name.
 {{- end -}}
 {{- end -}}
 
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "datadog.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
 {{/*
 Return secret name to be used based on provided values.
 */}}
@@ -40,30 +47,6 @@ Return secret name to be used based on provided values.
 {{- default $fullName .Values.datadog.appKeyExistingSecret | quote -}}
 {{- end -}}
 
-{{/*
-Create a default fully qualified confd name.
-We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
-*/}}
-{{- define "datadog.confd.fullname" -}}
-{{- printf "%s-datadog-confd" .Release.Name | trunc 63 | trimSuffix "-" -}}
-{{- end -}}
-
-{{/*
-Create a default fully qualified checksd name.
-We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
-*/}}
-{{- define "datadog.checksd.fullname" -}}
-{{- printf "%s-datadog-checksd" .Release.Name | trunc 63 | trimSuffix "-" -}}
-{{- end -}}
-
-{{/*
-Create a default fully qualified cluster-agent name.
-We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
-*/}}
-{{- define "datadog.clusterAgent.fullname" -}}
-{{- printf "%s-cluster-agent" .Release.Name | trunc 63 | trimSuffix "-" -}}
-{{- end -}}
-
 {{/*
 Return the appropriate apiVersion for RBAC APIs.
 */}}
diff --git a/stable/datadog/templates/agent-apiservice.yaml b/stable/datadog/templates/agent-apiservice.yaml
index 774e80be0..e3f69d1f2 100644
--- a/stable/datadog/templates/agent-apiservice.yaml
+++ b/stable/datadog/templates/agent-apiservice.yaml
@@ -10,7 +10,7 @@ metadata:
     release: {{ .Release.Name | quote }}
 spec:
   service:
-    name: {{ template "datadog.clusterAgent.fullname" . }}-metrics-api
+    name: {{ template "datadog.fullname" . }}-cluster-agent-metrics-api
     namespace: {{ .Release.Namespace }}
   version: v1beta1
   insecureSkipTLSVerify: true
diff --git a/stable/datadog/templates/agent-clusterrole.yaml b/stable/datadog/templates/agent-clusterrole.yaml
index 0a0871207..d6a8ba6f9 100644
--- a/stable/datadog/templates/agent-clusterrole.yaml
+++ b/stable/datadog/templates/agent-clusterrole.yaml
@@ -7,7 +7,7 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
     heritage: {{ .Release.Service | quote }}
     release: {{ .Release.Name | quote }}
-  name: {{ template "datadog.clusterAgent.fullname" . }}
+  name: {{ template "datadog.fullname" . }}-cluster-agent
 rules:
 - apiGroups:
   - ""
diff --git a/stable/datadog/templates/agent-clusterrolebinding-auth-delegator.yaml b/stable/datadog/templates/agent-clusterrolebinding-auth-delegator.yaml
index f173e098c..20e48daec 100644
--- a/stable/datadog/templates/agent-clusterrolebinding-auth-delegator.yaml
+++ b/stable/datadog/templates/agent-clusterrolebinding-auth-delegator.yaml
@@ -7,13 +7,13 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
-  name: {{ template "datadog.clusterAgent.fullname" . }}:system:auth-delegator
+  name: {{ template "datadog.fullname" . }}-cluster-agent:system:auth-delegator
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
   name: system:auth-delegator
 subjects:
   - kind: ServiceAccount
-    name: {{ template "datadog.clusterAgent.fullname" . }}
+    name: {{ template "datadog.fullname" . }}-cluster-agent
     namespace: {{ .Release.Namespace }}
 {{- end -}}
diff --git a/stable/datadog/templates/agent-clusterrolebinding.yaml b/stable/datadog/templates/agent-clusterrolebinding.yaml
index b231858fb..b525ceacd 100644
--- a/stable/datadog/templates/agent-clusterrolebinding.yaml
+++ b/stable/datadog/templates/agent-clusterrolebinding.yaml
@@ -7,13 +7,13 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
-  name: {{ template "datadog.clusterAgent.fullname" . }}
+  name: {{ template "datadog.fullname" . }}-cluster-agent
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
-  name: {{ template "datadog.clusterAgent.fullname" . }}
+  name: {{ template "datadog.fullname" . }}-cluster-agent
 subjects:
   - kind: ServiceAccount
-    name: {{ template "datadog.clusterAgent.fullname" . }}
+    name: {{ template "datadog.fullname" . }}-cluster-agent
     namespace: {{ .Release.Namespace }}
 {{- end -}}
diff --git a/stable/datadog/templates/agent-secret.yaml b/stable/datadog/templates/agent-secret.yaml
index 793eb05a7..7ee758c2d 100644
--- a/stable/datadog/templates/agent-secret.yaml
+++ b/stable/datadog/templates/agent-secret.yaml
@@ -3,7 +3,7 @@
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ template "datadog.clusterAgent.fullname" . }}
+  name: {{ template "datadog.fullname" . }}-cluster-agent
   labels:
     app: "{{ template "datadog.fullname" . }}"
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
diff --git a/stable/datadog/templates/agent-service-cmd.yaml b/stable/datadog/templates/agent-service-cmd.yaml
index 28f53c743..ba9a97299 100644
--- a/stable/datadog/templates/agent-service-cmd.yaml
+++ b/stable/datadog/templates/agent-service-cmd.yaml
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ template "datadog.clusterAgent.fullname" . }}
+  name: {{ template "datadog.fullname" . }}-cluster-agent
   labels:
     app: "{{ template "datadog.fullname" . }}"
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
@@ -11,7 +11,7 @@ metadata:
 spec:
   type: ClusterIP
   selector:
-    app: {{ template "datadog.clusterAgent.fullname" . }}
+    app: {{ template "datadog.fullname" . }}-cluster-agent
   ports:
   - port: 5005
     name: agentport
diff --git a/stable/datadog/templates/agent-service-metrics.yaml b/stable/datadog/templates/agent-service-metrics.yaml
index f43e338c7..b9869fa42 100644
--- a/stable/datadog/templates/agent-service-metrics.yaml
+++ b/stable/datadog/templates/agent-service-metrics.yaml
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ template "datadog.clusterAgent.fullname" . }}-metrics-api
+  name: {{ template "datadog.fullname" . }}-cluster-agent-metrics-api
   labels:
     app: "{{ template "datadog.fullname" . }}"
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
@@ -11,7 +11,7 @@ metadata:
 spec:
   type: ClusterIP
   selector:
-    app: {{ template "datadog.clusterAgent.fullname" . }}
+    app: {{ template "datadog.fullname" . }}-cluster-agent
   ports:
   - port: 443
     name: metricsapi
diff --git a/stable/datadog/templates/agent-serviceaccount.yaml b/stable/datadog/templates/agent-serviceaccount.yaml
index b15e3a8c5..375d3c563 100644
--- a/stable/datadog/templates/agent-serviceaccount.yaml
+++ b/stable/datadog/templates/agent-serviceaccount.yaml
@@ -7,5 +7,5 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
     heritage: {{ .Release.Service | quote }}
     release: {{ .Release.Name | quote }}
-  name: {{ template "datadog.clusterAgent.fullname" . }}
+  name: {{ template "datadog.fullname" . }}-cluster-agent
 {{- end -}}
diff --git a/stable/datadog/templates/agenthpa-rolebinding.yaml b/stable/datadog/templates/agenthpa-rolebinding.yaml
index eaf2758e7..2e3be3c1e 100644
--- a/stable/datadog/templates/agenthpa-rolebinding.yaml
+++ b/stable/datadog/templates/agenthpa-rolebinding.yaml
@@ -7,13 +7,13 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
-  name: "{{ template "datadog.clusterAgent.fullname" . }}"
+  name: "{{ template "datadog.fullname" . }}-cluster-agent"
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
   name: extension-apiserver-authentication-reader
 subjects:
   - kind: ServiceAccount
-    name: {{ template "datadog.clusterAgent.fullname" . }}
+    name: {{ template "datadog.fullname" . }}-cluster-agent
     namespace: {{ .Release.Namespace }}
 {{- end -}}
diff --git a/stable/datadog/templates/checksd-configmap.yaml b/stable/datadog/templates/checksd-configmap.yaml
index 068587620..2d4f50fe3 100644
--- a/stable/datadog/templates/checksd-configmap.yaml
+++ b/stable/datadog/templates/checksd-configmap.yaml
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: {{ template "datadog.checksd.fullname" . }}
+  name: {{ template "datadog.fullname" . }}-checksd
   labels:
     app: "{{ template "datadog.fullname" . }}"
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
diff --git a/stable/datadog/templates/cluster-agent-deployment.yaml b/stable/datadog/templates/cluster-agent-deployment.yaml
index 5de9de64c..2a54a9941 100644
--- a/stable/datadog/templates/cluster-agent-deployment.yaml
+++ b/stable/datadog/templates/cluster-agent-deployment.yaml
@@ -2,7 +2,7 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ template "datadog.clusterAgent.fullname" . }}
+  name: {{ template "datadog.fullname" . }}-cluster-agent
   labels:
     app: "{{ template "datadog.fullname" . }}"
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
@@ -12,12 +12,12 @@ spec:
   replicas: {{ .Values.clusterAgent.replicas }}
   selector:
     matchLabels:
-      app: {{ template "datadog.clusterAgent.fullname" . }}
+      app: {{ template "datadog.fullname" . }}-cluster-agent
   template:
     metadata:
       labels:
-        app: {{ template "datadog.clusterAgent.fullname" . }}
-      name: {{ template "datadog.clusterAgent.fullname" . }}
+        app: {{ template "datadog.fullname" . }}-cluster-agent
+      name: {{ template "datadog.fullname" . }}-cluster-agent
     spec:
       {{- if .Values.clusterAgent.image.pullSecrets }}
       imagePullSecrets:
@@ -75,12 +75,12 @@ spec:
           - name: DD_COLLECT_KUBERNETES_EVENTS
             value: {{ .Values.datadog.collectEvents | quote}}
           - name: DD_CLUSTER_AGENT_KUBERNETES_SERVICE_NAME
-            value: {{ template "datadog.clusterAgent.fullname" . }}
+            value: {{ template "datadog.fullname" . }}-cluster-agent
           {{- end }}
           - name: DD_CLUSTER_AGENT_AUTH_TOKEN
             valueFrom:
               secretKeyRef:
-                name: {{ template "datadog.clusterAgent.fullname" . }}
+                name: {{ template "datadog.fullname" . }}-cluster-agent
                 key: token
           - name: DD_KUBE_RESOURCES_NAMESPACE
             value: {{ .Release.Namespace }}
@@ -115,5 +115,5 @@ spec:
       affinity:
 {{ toYaml .Values.clusterAgent.affinity | indent 8 }}
       {{- end }}
-      serviceAccountName: {{ if .Values.rbac.create }}{{ template "datadog.clusterAgent.fullname" . }}{{ else }}"{{ .Values.rbac.serviceAccountName }}"{{ end }}
+      serviceAccountName: {{ if .Values.rbac.create }}{{ template "datadog.fullname" . }}-cluster-agent{{ else }}"{{ .Values.rbac.serviceAccountName }}"{{ end }}
 {{ end }}
diff --git a/stable/datadog/templates/confd-configmap.yaml b/stable/datadog/templates/confd-configmap.yaml
index be9c9a513..98eebe050 100644
--- a/stable/datadog/templates/confd-configmap.yaml
+++ b/stable/datadog/templates/confd-configmap.yaml
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: {{ template "datadog.confd.fullname" . }}
+  name: {{ template "datadog.fullname" . }}-confd
   labels:
     app: "{{ template "datadog.fullname" . }}"
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
diff --git a/stable/datadog/templates/daemonset.yaml b/stable/datadog/templates/daemonset.yaml
index 45452b4ee..b8810a9bf 100644
--- a/stable/datadog/templates/daemonset.yaml
+++ b/stable/datadog/templates/daemonset.yaml
@@ -129,11 +129,11 @@ spec:
           - name: DD_CLUSTER_AGENT_ENABLED
             value: {{ .Values.clusterAgent.enabled | quote }}
           - name: DD_CLUSTER_AGENT_KUBERNETES_SERVICE_NAME
-            value: {{ template "datadog.clusterAgent.fullname" . }}
+            value: {{ template "datadog.fullname" . }}-cluster-agent
           - name: DD_CLUSTER_AGENT_AUTH_TOKEN
             valueFrom:
               secretKeyRef:
-                name: {{ template "datadog.clusterAgent.fullname" . }}
+                name: {{ template "datadog.fullname" . }}-cluster-agent
                 key: token
           {{- end }}
           - name: KUBERNETES
@@ -251,12 +251,12 @@ spec:
         {{- if (or (.Values.datadog.confd) (.Values.datadog.autoconf)) }}
         - name: confd
           configMap:
-            name: {{ template "datadog.confd.fullname" . }}
+            name: {{ template "datadog.fullname" . }}-confd
         {{- end }}
         {{- if .Values.datadog.checksd }}
         - name: checksd
           configMap:
-            name: {{ template "datadog.checksd.fullname" . }}
+            name: {{ template "datadog.fullname" . }}-checksd
         {{- end }}
         {{- if .Values.datadog.logsEnabled }}
         - hostPath:
diff --git a/stable/datadog/templates/deployment.yaml b/stable/datadog/templates/deployment.yaml
index 7ab7e44a6..a2b3710e6 100644
--- a/stable/datadog/templates/deployment.yaml
+++ b/stable/datadog/templates/deployment.yaml
@@ -160,12 +160,12 @@ spec:
         {{- if (or (.Values.datadog.confd) (.Values.datadog.autoconf)) }}
         - name: confd
           configMap:
-            name: {{ template "datadog.confd.fullname" . }}
+            name: {{ template "datadog.fullname" . }}-confd
         {{- end }}
         {{- if .Values.datadog.checksd }}
         - name: checksd
           configMap:
-            name: {{ template "datadog.checksd.fullname" . }}
+            name: {{ template "datadog.fullname" . }}-checksd
         {{- end }}
 {{- if .Values.datadog.volumes }}
 {{ toYaml .Values.datadog.volumes | indent 8 }}
diff --git a/stable/datadog/templates/hpa-clusterrole.yaml b/stable/datadog/templates/hpa-clusterrole.yaml
index b868cff60..dddfd45fc 100644
--- a/stable/datadog/templates/hpa-clusterrole.yaml
+++ b/stable/datadog/templates/hpa-clusterrole.yaml
@@ -7,7 +7,7 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
     heritage: {{ .Release.Service | quote }}
     release: {{ .Release.Name | quote }}
-  name: {{ template "datadog.clusterAgent.fullname" . }}-external-metrics-reader
+  name: {{ template "datadog.fullname" . }}-cluster-agent-external-metrics-reader
 rules:
 - apiGroups:
   - "external.metrics.k8s.io"
diff --git a/stable/datadog/templates/hpa-clusterrolebinding.yaml b/stable/datadog/templates/hpa-clusterrolebinding.yaml
index 4108746d2..7b14c3a7a 100644
--- a/stable/datadog/templates/hpa-clusterrolebinding.yaml
+++ b/stable/datadog/templates/hpa-clusterrolebinding.yaml
@@ -7,11 +7,11 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
     release: {{ .Release.Name | quote }}
     heritage: {{ .Release.Service | quote }}
-  name: {{ template "datadog.clusterAgent.fullname" . }}-external-metrics-reader
+  name: {{ template "datadog.fullname" . }}-cluster-agent-external-metrics-reader
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
-  name: {{ template "datadog.clusterAgent.fullname" . }}-external-metrics-reader
+  name: {{ template "datadog.fullname" . }}-cluster-agent-external-metrics-reader
 subjects:
 - kind: ServiceAccount
   name: horizontal-pod-autoscaler
diff --git a/stable/datadog/values.yaml b/stable/datadog/values.yaml
index e7a8d0987..02af14bc8 100644
--- a/stable/datadog/values.yaml
+++ b/stable/datadog/values.yaml
@@ -32,6 +32,9 @@ image:
   #  pullSecrets:
   #    - name: "<REG_SECRET>"
 
+nameOverride: ""
+fullnameOverride: ""
+
 datadog:
 
   ## @param apiKey - string - required
